src/sql/workbench/contrib/query/test/browser/queryActions.test.ts

```diff
@@ -560,6 +560,119 @@ suite('SQL QueryAction Tests', () => {
 		// ... The connection should have changed to the provided database
 		assert.equal(listItem.currentDatabaseName, eventParams.connectionProfile.databaseName);
 	});
+
+	test('runCurrent - opens connection dialog when there are no active connections', async () => {
+		// setting up environment
+		let isConnected = false;
+		let predefinedSelection: IRange = { startLineNumber: 1, startColumn: 1, endLineNumber: 1, endColumn: 1 };
+		let calledRunQueryStatementOnInput: boolean = undefined;
+
+		// setting up assert variables
+		let countCalledShowDialog: number = 0;
+
+		// mocking query editor
+		const contextkeyservice = new MockContextKeyService();
+		let queryEditor = TypeMoq.Mock.ofType(QueryEditor, TypeMoq.MockBehavior.Loose, undefined, new TestThemeService(),
+			new TestStorageService(), contextkeyservice, undefined, new TestFileService(), undefined);
+		queryEditor.setup(x => x.input).returns(() => testQueryInput.object);
+		queryEditor.setup(x => x.getSelection(false)).returns(() => { return predefinedSelection; });
+
+		// Mocking runQueryStatment in unititledQueryEditorInput
+		testQueryInput.setup(x => x.runQueryStatement(TypeMoq.It.isAny())).callback(() => { calledRunQueryStatementOnInput = true; });
+
+		// mocking isConnected in ConnectionManagementService
+		connectionManagementService.setup(x => x.isConnected(TypeMoq.It.isAnyString())).returns(() => isConnected);
+
+		// mocking showConnectionDialog in ConnectionManagementService
+		connectionManagementService.setup(x => x.showConnectionDialog(TypeMoq.It.isAny())).callback((params: INewConnectionParams) => {
+			countCalledShowDialog++;
+		});
+
+		// mocking QueryModelService
+		let queryModelService = TypeMoq.Mock.ofType(QueryModelService, TypeMoq.MockBehavior.Loose);
+		queryModelService.setup(x => x.runQueryStatement(TypeMoq.It.isAny(), TypeMoq.It.isAny()));
+
+		// Calling runCurrent with no open connection
+		let queryAction: RunQueryAction = new RunQueryAction(queryEditor.object, queryModelService.object, connectionManagementService.object);
+		calledRunQueryStatementOnInput = false;
+		await queryAction.runCurrent();
+
+		//connection dialog should open and runQueryStatement should not be called
+		assert.equal(calledRunQueryStatementOnInput, false, 'runCurrent should not call runQueryStatement');
+		testQueryInput.verify(x => x.runQueryStatement(TypeMoq.It.isAny()), TypeMoq.Times.never());
+		assert.equal(countCalledShowDialog, 1, 'showDialog is called');
```

Comment:
The message should indicate what went wrong - in this case you expected it to be called but it wasn't.